plugin/cmd/kube-scheduler/scheduler.go

```diff
@@ -36,11 +39,42 @@ func main() {
 	s := app.NewSchedulerServer()
 	s.AddFlags(pflag.CommandLine)
 
+	haconfig := ha.Config{}
+	haconfig.AddFlags(pflag.CommandLine)
+
 	util.InitFlags()
 	util.InitLogs()
 	defer util.FlushLogs()
 
 	verflag.PrintAndExitIfRequested()
 
-	s.Run(pflag.CommandLine.Args())
+	startSched := func(leaseUserInfo *ha.LeaseUser) bool {
+		leaseUserInfo.Running = true
+		glog.Infof("Starting kube scheduler. %v", leaseUserInfo)
+		if err := s.Run(pflag.CommandLine.Args()); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			glog.Infof("EXITING NOW ! Killed")
+			os.Exit(1)
```

Comment:
I think this should it be exit 0.. not sure though.
Also.. since go programs Exit on Return.. I wonder wether we should just return rather than hardcode the exit?